### ISSUE TITLE
`Colorbar` breaks with min limit == max limit

### ISSUE THREAD

**Issue by eliascarv** [MakieOrg/Makie.jl#3174]
Code to reproduce the error:
```julia
julia> using GLMakie

julia> fig = Figure()

julia> Axis(fig[1, 1])
Axis with 0 plots:


julia> Colorbar(fig[1, 2], limits = (1, 1))
ERROR: ArgumentError: range step cannot be zero
```

**Comment by lazarusA** [comment]
what would be the expected output? a colorbar with one color (a categorical cmap)?

**Comment by eliascarv** [comment]
Yes, I was expecting a colormap with just one color and one value.

**Comment by jkrumbiegel** [comment]
colors are interpolated by scaling the colorrange to 0, 1 and mapping that to the colorscheme's colors. If you have a colorrange of width 0, you can't look up any value in the colorscheme

**Comment by jkrumbiegel** [comment]
The error message is not good though

**Comment by eliascarv** [comment]
Maybe, for this corner case, generate a Colorbar with the first color of the color scheme or without color just with the border and the value. Another option would be to display a warning and show an empty Colorbar or display a warning and not show the Colorbar.

**Comment by jkrumbiegel** [comment]
I'm pretty against warnings, they make it hard to let code run automatically, on CI, etc. Often people don't see these things and then wonder later why some colorbar is empty. Also I don't really like doing the "next best thing" if behavior is essentially undefined. I'd go for a clearer error here. Errors stop people in their tracks and make them question what they're doing :)

**Comment by eliascarv** [comment]
I agree. A clearer error message seems to be the best way forward.

**Comment by ffreyer** [comment]
This works on master, producing an empty Colorbar (no ticks, no colors, just a rect). Updating the limits to a valid range produces ticks and colors without issues. Going back to `(1, 1)` reverts back to the empty Colorbar without issues.

Probably fixed by #5682
